UMS/man/man3/restoreConfiguration.3
'\" t
.\"     Title: restoreConfiguration
.\"    Author:  SysFera SAS <info@sysfera.com>
.\" Generator: DocBook XSL Stylesheets v1.75.2 <http://docbook.sf.net/>
.\"      Date: February 2011
.\"    Manual: C++ API Reference
.\"    Source: VISHNU 1.0
.\"  Language: English
.\"
.TH "RESTORECONFIGURATION" "3" "February 2011" "VISHNU 1.0" "C++ API Reference"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
restoreConfiguration \- restores the configuration of VISHNU
.SH "SYNOPSIS"
.HP \w'int\ vishnu::restoreConfiguration('u
.BI "int vishnu::restoreConfiguration(const\ string&\ " "sessionKey" ", const\ string&\ " "filePath" ");"
.SH "DESCRIPTION"
.PP
This function must be used carefully as it replaces all the content of the VISHNU central database with the information stored in the provided file\&. This file contains the list of users, the lists of machines and the list of local user configurations\&. It can be created using the vishnu_save_configuration command\&. The "root" VISHNU user is the only user authorized to call this function, and this action must be done without any other user connected to VISHNU\&. After restoring, the vishnu database is re\-initialized\&.
.SH "ARGUMENTS"
.PP
\fIsessionKey\fR
.RS 4
Input argument\&. The sessionKey is the encrypted identifier of the session generated by VISHNU\&.
.RE
.PP
\fIfilePath\fR
.RS 4
Input argument\&. The filePath is the path of the file used to restore VISHNU configuration\&.
.RE
.SH "EXCEPTIONS"
.PP
The following exceptions may be thrown:
.PP
"Internal Error: SysDS service failure" [1]
.RS 4
.RE
.PP
"Internal Error: Database returned an error" [2]
.RS 4
.RE
.PP
"Internal Error: Database connection failure" [3]
.RS 4
.RE
.PP
"Internal Error: System call failure" [4]
.RS 4
.RE
.PP
"Internal Error: Undefined exception" [9]
.RS 4
.RE
.PP
"The user is not an administrator" [25]
.RS 4
.RE
.PP
"A problem occurs during the configuration restoring" [40]
.RS 4
.RE
.SH "AUTHOR"
.PP
\fB SysFera SAS\fR <\&info@sysfera.com\&>
.RS 4
Developed the first version of these man pages as part of the VISHNU project.
.RE
.SH "COPYRIGHT"
.br
Copyright \(co 2011 SysFera SAS
.br
.PP
These manual pages are provided under the following conditions:
.PP
Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
.RE
.PP
This software is governed by the CECILL licence under French law and abiding by the rules of distribution of free software. You can use, modify and/ or redistribute the software under the terms of the CeCILL license as circulated by CEA, CNRS and INRIA at the following URL "http://www.cecill.info".
.PP
As a counterpart to the access to the source code and rights to copy, modify and redistribute granted by the license, users are provided only with a limited warranty and the software's author, the holder of the economic rights, and the successive licensors have only limited liability.
.PP
In this respect, the user's attention is drawn to the risks associated with loading, using, modifying and/or developing or reproducing the software by the user in light of its specific status of free software, that may mean that it is complicated to manipulate, and that also therefore means that it is reserved for developers and experienced professionals having in-depth computer knowledge. Users are therefore encouraged to load and test the software's suitability as regards their requirements in conditions enabling the security of their systems and/or data to be ensured and, more generally, to use and operate it in the same conditions as regards security.
.sp


UMS/src/cli/restoreConfiguration.cc
#include "restoreConfiguration.hh"
#include "utils.hh"
namespace po = boost::program_options;

using namespace std;

int main (int ac, char* av[]){


	string filePath;

	string sessionKey;

	string dietConfig;

		int reqParam=0;   // to count the required parameters for the command

/**************** Describe options *************/


		Options opt(av[0] );

		opt.add("version,v",
				"print version message",
				GENERIC);

        opt.add("dietConfig,c",
				        "The diet config file",
						ENV,
						dietConfig);

				opt.add("sessionKey,s",
						           "The session key",
											 ENV,
											 sessionKey);

		opt.add("filePath",
					  "The path of the VISHNU configuration file",
						HIDDEN,
						filePath);

		opt.setPosition("filePath",-1);


		try {

/**************  Parse to retrieve option values  ********************/

		opt.parse_cli(ac,av);

		//opt.parse_cfile();

		opt.parse_env(env_name_mapper());

		opt.notify();





/********  Process **************************/

		if (opt.count("filePath")){

			cout <<"The file Path is " << filePath << endl;

			reqParam=reqParam+1;
		}



		checkVishnuConfig(opt);

		if(opt.count("sessionKey")){

			cout << "The session Key is: " << sessionKey <<endl;

		}


		 if ((reqParam < RCPARAM)|| (opt.count("help"))){

			 cout << "Usage: " << av[0] <<" filePath"<<endl;

			 cout << opt << endl;

			 return 0;
		 }




/************** Call UMS connect service *******************************/


               // initializing DIET
              if (diet_initialize(dietConfig.c_str(), ac, av)) {

				  cerr << "DIET initialization failed !" << endl;

				  return 1;
              }

              restoreConfiguration(sessionKey, filePath);



	}// End of try bloc

	catch(std::exception& e){
		cout << e.what() <<endl;
		return 1;
	}

	return 0;

}// end of main


